engine/src/main/java/org/terasology/engine/rendering/logic/SkeletonRenderer.java

```diff
@@ -340,10 +340,15 @@ public void renderOverlay() {
             int index = 0;
             for (EntityRef entity : entityManager.getEntitiesWith(SkeletalMeshComponent.class, LocationComponent.class)) {
                 SkeletalMeshComponent skeletalMesh = entity.getComponent(SkeletalMeshComponent.class);
+                LocationComponent locationComponent = entity.getComponent(LocationComponent.class);
                 if (skeletalMesh.boneEntities == null) {
                     continue;
                 }
 
+                Vector3f location = locationComponent.getWorldPosition(new Vector3f());
+                Quaternionf rotation = locationComponent.getWorldRotation(new Quaternionf());
+                Matrix4f transform = new Matrix4f().translationRotateScale(location, rotation, 1.0f);
```

Comment:
Is there any reason to create new instances for the location vector, the rotation quaternion, and the transformation matrix with each render call for each entity?

You often raise concerns about code like this, so I'm wondering about all the instantiations in this code render code block that gets called quite often (if the debug feature is enabled).

Reply:
umm, I guess I could use a temporary Matrix. this is because we have custom offset and translation for the SkeletonComponent :/

Reply:
So we need individual vectors here and cannot reuse a single temporary object instance for rendering the skeleton for all entities?

Reply:
nope, we can cache it, I was just getting lazy. was looking at this at 10 something.